publish/npm/install.js

```diff
@@ -0,0 +1,402 @@
+    return false;
+  }
+}
+
+// Get version from GitHub environment variable or fallback to package.json
+function getVersion() {
+  // Try to get version from GitHub environment variables
+  // GITHUB_REF_NAME contains just the tag name (e.g., "v1.0.0")
+  // GITHUB_REF contains the full ref (e.g., "refs/tags/v1.0.0")
+  let version = process.env.GITHUB_REF_NAME || process.env.GITHUB_REF;
+
+  if (version) {
+    // If GITHUB_REF, extract tag name from "refs/tags/v1.0.0"
+    if (version.startsWith("refs/tags/")) {
+      version = version.replace("refs/tags/", "");
+    }
+
+    // Ensure version starts with 'v'
+    if (!version.startsWith("v")) {
+      version = `v${version}`;
+    }
+
+    console.log(`📌 Using version from GitHub environment: ${version}`);
+    return version;
+  }
+
+  // Fallback to package.json
+  try {
+    const packageJson = JSON.parse(
+      fs.readFileSync(path.join(__dirname, "package.json"), "utf8"),
+    );
+    const fallbackVersion = `v${packageJson.version}`;
+    console.log(
+      `📌 Using fallback version from package.json: ${fallbackVersion}`,
+    );
+    return fallbackVersion;
+  } catch (error) {
+    throw new Error(
+      `Failed to read version from environment or package.json: ${error.message}`,
+    );
+  }
+}
+
+// Get release info for specific version from GitHub API
+function getReleaseByTag(tag) {
+  const releaseUrl = `https://api.github.com/repos/safedep/pmg/releases/tags/${tag}`;
+
+  return new Promise((resolve, reject) => {
+    https
+      .get(
+        releaseUrl,
+        {
+          headers: {
+            "User-Agent": "pmg-npm-installer",
+          },
+        },
+        (response) => {
+          let data = "";
+
+          response.on("data", (chunk) => {
+            data += chunk;
+          });
+
+          response.on("end", () => {
+            try {
+              if (response.statusCode === 404) {
+                reject(new Error(`Release ${tag} not found`));
+                return;
+              }
+              if (response.statusCode !== 200) {
+                reject(
+                  new Error(
+                    `Failed to fetch release info: ${response.statusCode}`,
+                  ),
+                );
+                return;
+              }
+              const release = JSON.parse(data);
+              resolve(release);
+            } catch (err) {
+              reject(new Error(`Failed to parse release info: ${err.message}`));
+            }
+          });
+        },
+      )
+      .on("error", reject);
+  });
+}
+
+async function install() {
+  try {
+    console.log("📦 Installing PMG binary...");
+
+    // Get platform info
+    const platformInfo = getPlatformInfo();
+    console.log(`🔍 Detected platform: ${process.platform}-${process.arch}`);
+    console.log(`📋 Looking for: ${platformInfo.filename}`);
+
+    // Get version from GitHub environment variable or fallback to package.json
+    const version = getVersion();
```

Comment:
I don't think we should dynamically resolve version. This opens up to supply chain attacks. `@safedep/pmg` version X should only install version X of PMG binary. Since we plan to publish `npm` wrapper with every PMG release, we should ensure if that I do: `npm install @safedep/pmg@1.2.3`, I also get PMG `1.2.3`.

What do you think?